nodeadm/internal/kubelet/config.go

```diff
@@ -249,9 +249,14 @@ func (ksc *kubeletConfig) withCloudProvider(kubeletVersion string, cfg *api.Node
 		flags["cloud-provider"] = "external"
 		// provider ID needs to be specified when the cloud provider is external
 		ksc.ProviderID = ptr.String(getProviderId(cfg.Status.Instance.AvailabilityZone, cfg.Status.Instance.ID))
-		// the name of the Node object must equal the EC2 PrivateDnsName
-		// see: https://github.com/awslabs/amazon-eks-ami/pull/1264
-		flags["hostname-override"] = cfg.Status.Instance.PrivateDNSName
+		if cfg.Spec.Instance.NodeName.Strategy == api.InstanceIDName {
+			zap.L().Info("Opt-in Instance Id naming strategy")
+			flags["hostname-override"] = cfg.Status.Instance.ID
+		} else {
+			// the name of the Node object default to EC2 PrivateDnsName
+			// see: https://github.com/awslabs/amazon-eks-ami/pull/1264
+			flags["hostname-override"] = cfg.Status.Instance.PrivateDNSName
```

Comment:
should we also remove the api call to propogate the private dns name when nodename strategy is instanceID?